core/src/main/java/apoc/merge/Merge.java

```diff
@@ -31,12 +35,37 @@
     }
 
     @Procedure(value="apoc.merge.node", mode = Mode.WRITE)
-    @Description("\"apoc.merge.node.eager(['Label'], identProps:{key:value, ...}, onCreateProps:{key:value,...}, onMatchProps:{key:value,...}}) - merge nodes with dynamic labels, with support for setting properties ON CREATE or ON MATCH")
+    @Description("\"apoc.merge.node(['Label'], identProps:{key:value, ...}, onCreateProps:{key:value,...}, onMatchProps:{key:value,...}}) - merge nodes with dynamic labels, with support for setting properties ON CREATE or ON MATCH")
     public Stream<NodeResult> nodes(@Name("label") List<String> labelNames,
-                                        @Name("identProps") Map<String, Object> identProps,
-                                        @Name(value = "props",defaultValue = "{}") Map<String, Object> props,
-                                        @Name(value = "onMatchProps",defaultValue = "{}") Map<String, Object> onMatchProps) {
-        if (identProps==null || identProps.isEmpty()) {
+                                          @Name("identProps") Map<String, Object> identProps,
+                                          @Name(value = "props",defaultValue = "{}") Map<String, Object> props,
+                                          @Name(value = "onMatchProps",defaultValue = "{}") Map<String, Object> onMatchProps) {
+        final Result nodeResult = getNodeResult(labelNames, identProps, props, onMatchProps);
+        return nodeResult.columnAs("n").stream().map(node -> new NodeResult((Node) node));
+    }
+
+    @Procedure(value="apoc.merge.nodeWithStats.eager", mode = Mode.WRITE, eager = true)
+    @Description("apoc.merge.nodeWithStats.eager - same as apoc.merge.node.eager providing queryStatistics into result")
+    public Stream<NodeResultWithStats> nodeWithStatsEager(@Name("label") List<String> labelNames,
```

Comment:
Should we make the stats and eager as config we can set on the procedure? Or does that complicate/overload each proc and testing too much?

i.e. apoc.merge.node(['Label'], identProps:{key:value, ...}, onCreateProps:{key:value,...}, onMatchProps:{key:value,...}, config: {eager: true, stats: true})

Reply:
Regarding the `eager`, the procedure must start in a method with a`@Procedure(... eager = true)`.
Perhaps we could change somehow the annotation parameter at runtime, but I don't think it's worth it.

With `stats`, I should change the `apoc.merge.node` result to `Stream<NodeResultWithStats>`,
but in this case the `apoc.merge.node` (without start) will yield also a column `stats` with value null.
Maybe it is better to leave it unchanged?

Reply:
Ah, ok. Never mind. Thanks for clarifying!